src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_eckey.c

```diff
@@ -0,0 +1,193 @@
+                    CheckJNIExceptions(env); // The destroy call might throw an exception. Clear the exception state.
+                }
+            }
+
+            ReleaseGRef(env, r->keyPair);
+            ReleaseGRef(env, r->curveParameters);
+            free(r);
+        }
+    }
+}
+
+EC_KEY* AndroidCryptoNative_EcKeyCreateByOid(const char* oid)
+{
+    JNIEnv* env = GetJNIEnv();
+
+    // Older versions of Android don't support mapping an OID to a curve name,
+    // so do some of the common mappings here.
+    jstring oidStr;
+    if (strcmp(oid, "1.3.132.0.33") == 0)
+    {
+        oidStr = JSTRING("secp224r1");
+    }
+    else if (strcmp(oid, "1.3.132.0.34") == 0 || strcmp(oid, "nistP384") == 0)
+    {
+        oidStr = JSTRING("secp384r1");
+    }
+    else if (strcmp(oid, "1.3.132.0.35") == 0 || strcmp(oid, "nistP521") == 0)
+    {
+        oidStr = JSTRING("secp521r1");
+    }
+    else if (strcmp(oid, "1.2.840.10045.3.1.7") == 0 || strcmp(oid, "nistP256") == 0)
+    {
+        oidStr = JSTRING("secp256r1");
+    }
+    else 
+    {
+        oidStr = JSTRING(oid);
+    }
+    jstring ec = JSTRING("EC");
+
+    // First, generate the key pair based on the curve defined by the oid.
+    jobject paramSpec = (*env)->NewObject(env, g_ECGenParameterSpecClass, g_ECGenParameterSpecCtor, oidStr);
+    ReleaseLRef(env, oidStr);
+
+    jobject keyPairGenerator =
+        (*env)->CallStaticObjectMethod(env, g_keyPairGenClass, g_keyPairGenGetInstanceMethod, ec);
+    (*env)->CallVoidMethod(env, keyPairGenerator, g_keyPairGenInitializeWithParamsMethod, paramSpec);
+
+    ReleaseLRef(env, paramSpec);
+    if (CheckJNIExceptions(env))
+    {
+        LOG_DEBUG("Failed to create curve");
+        ReleaseLRef(env, ec);
+        ReleaseLRef(env, keyPairGenerator);
+        return NULL;
+    }
+
+    jobject keyPair = (*env)->CallObjectMethod(env, keyPairGenerator, g_keyPairGenGenKeyPairMethod);
+
+    ReleaseLRef(env, keyPairGenerator);
+
+    // Now that we have the key pair, we can get the curve parameters from the public key.
+    jobject keyFactory = (*env)->CallStaticObjectMethod(env, g_KeyFactoryClass, g_KeyFactoryGetInstanceMethod, ec);
+    jobject publicKey = (*env)->CallObjectMethod(env, keyPair, g_keyPairGetPublicMethod);
+    jobject keySpec =
+        (*env)->CallObjectMethod(env, keyFactory, g_KeyFactoryGetKeySpecMethod, publicKey, g_ECPublicKeySpecClass);
+
+    ReleaseLRef(env, ec);
+    ReleaseLRef(env, publicKey);
+    ReleaseLRef(env, keyFactory);
+
+    if (CheckJNIExceptions(env))
+    {
+        ReleaseLRef(env, keySpec);
+        ReleaseLRef(env, keyPair);
+        return NULL;
+    }
+
+    jobject curveParameters = (*env)->CallObjectMethod(env, keySpec, g_ECPublicKeySpecGetParams);
+
+    jobject group = (*env)->CallObjectMethod(env, curveParameters, g_ECParameterSpecGetCurve);
+
+    jobject field = (*env)->CallObjectMethod(env, group, g_EllipticCurveGetField);
+
+    ReleaseLRef(env, field);
+    ReleaseLRef(env, group);
```

Comment:
Are these lines still needed?  If there's a side effect from calling them, please add a comment so that the code doesn't get overzealously cleaned up.